src/peft/tuners/lora.py

```diff
@@ -337,15 +369,18 @@ def add_weighted_adapter(self, adapters, weights, adapter_name):
         for key in key_list:
             _, target, _ = _get_submodules(self.model, key)
             if isinstance(target, LoraLayer):
-                target.lora_A[adapter_name].weight.data = target.lora_A[adapter_name].weight.data * 0.0
-                target.lora_B[adapter_name].weight.data = target.lora_B[adapter_name].weight.data * 0.0
-                for adapter, weight in zip(adapters, weights):
-                    if adapter not in target.lora_A:
-                        continue
-                    target.lora_A[adapter_name].weight.data += (
-                        target.lora_A[adapter].weight.data * weight * target.scaling[adapter]
-                    )
-                    target.lora_B[adapter_name].weight.data += target.lora_B[adapter].weight.data * weight
+                if adapter_name in target.lora_A:
+                    target.lora_A[adapter_name].weight.data = target.lora_A[adapter_name].weight.data * 0.0
+                    target.lora_B[adapter_name].weight.data = target.lora_B[adapter_name].weight.data * 0.0
+                    for adapter, weight in zip(adapters, weights):
+                        if adapter not in target.lora_A:
+                            continue
+                        target.lora_A[adapter_name].weight.data += (
+                            target.lora_A[adapter].weight.data * weight * target.scaling[adapter]
+                        )
+                        target.lora_B[adapter_name].weight.data += target.lora_B[adapter].weight.data * weight
+                else:
+                    raise ValueError("Cannot implement this function in Embedding Layer")
```

Comment:
why can't this be supported? We could do the same as above, can't we?

Reply:
I initially thought that adapting the Embedding Layer would be different since it is based on nn.Parameter implementation. However, after considering your input, I realized that following the approach used above should work. I have implemented this change and committed it.